src/contracts/core/AVSDirectory.sol

```diff
@@ -288,36 +343,71 @@ contract AVSDirectory is
         override
         onlyWhenNotPaused(PAUSED_OPERATOR_REGISTER_DEREGISTER_TO_AVS)
     {
-        MemberInfo storage member = memberInfo[msg.sender][operator];
-
-        // Assert `member.isLegacyOperator` is actively registered to the (caller) avs.
-        require(member.isLegacyOperator, "AVSDirectory.deregisterOperatorFromAVS: operator not registered");
+        require(
+            avsOperatorStatus[msg.sender][operator] == OperatorAVSRegistrationStatus.REGISTERED,
+            "AVSDirectory.deregisterOperatorFromAVS: operator not registered"
+        );
 
-        // Mutate `member.isLegacyOperator` to `false`.
-        member.isLegacyOperator = false;
+        // Set the operator as deregistered
+        avsOperatorStatus[msg.sender][operator] = OperatorAVSRegistrationStatus.UNREGISTERED;
 
-        emit OperatorAVSRegistrationStatusUpdated(operator, msg.sender, false);
+        emit OperatorAVSRegistrationStatusUpdated(operator, msg.sender, OperatorAVSRegistrationStatus.UNREGISTERED);
     }
 
     /**
-     *  @notice Called by an AVS to emit an `AVSMetadataURIUpdated` event indicating the information has updated.
      *
-     *  @param metadataURI The URI for metadata associated with an AVS.
+     *                         INTERNAL FUNCTIONS
      *
-     *  @dev Note that the `metadataURI` is *never stored* and is only emitted in the `AVSMetadataURIUpdated` event.
      */
-    function updateAVSMetadataURI(string calldata metadataURI) external override {
-        emit AVSMetadataURIUpdated(msg.sender, metadataURI);
+
+    /**
+     * @notice Helper function used by migration & registration functions to register an operator to operator sets.
+     * @param avs The AVS that the operator is registering to.
+     * @param operator The operator to register.
+     * @param operatorSetIds The IDs of the operator sets.
+     */
+    function _registerToOperatorSets(address avs, address operator, uint32[] calldata operatorSetIds) internal {
+        // Loop over `operatorSetIds` array and register `operator` for each item.
+        for (uint256 i = 0; i < operatorSetIds.length; ++i) {
+            require(
+                isOperatorSet[avs][operatorSetIds[i]],
+                "AVSDirectory._registerOperatorToOperatorSets: invalid operator set"
+            );
+
+            // Assert `operator` has not already been registered to `operatorSetIds[i]`.
+            require(
+                !isMember[avs][operator][operatorSetIds[i]],
+                "AVSDirectory._registerOperatorToOperatorSets: operator already registered to operator set"
+            );
+
+            // Mutate `isMember` to `true`.
+            isMember[avs][operator][operatorSetIds[i]] = true;
```

Comment:
prefer `isOperatorInOperatorSet`

Reply:
idc tho, change the view function in the spec if u prefer this tho

Reply:
cc @0xClandestine on naming finalization - remember you were playing around with some ideas here